plugin-server/src/cdp/hog-function-manager.ts

```diff
@@ -89,66 +127,77 @@ export class HogFunctionManager {
         }
     }
 
-    public fetchHogFunction(id: HogFunctionType['id']): Promise<HogFunctionType | null> {
-        return fetchHogFunction(this.postgres, id)
+    public async fetchHogFunction(id: HogFunctionType['id']): Promise<HogFunctionType | null> {
+        const items: HogFunctionType[] = (
+            await this.postgres.query(
+                PostgresUse.COMMON_READ,
+                `SELECT ${HOG_FUNCTION_FIELDS.join(', ')}
+                FROM posthog_hogfunction
+                WHERE id = $1 AND deleted = FALSE`,
+                [id],
+                'fetchHogFunction'
+            )
+        ).rows
+        await this.enrichWithIntegrations(items)
+        return items[0] ?? null
     }
-}
-
-const HOG_FUNCTION_FIELDS = ['id', 'team_id', 'name', 'enabled', 'inputs', 'filters', 'bytecode']
 
-async function fetchAllHogFunctionsGroupedByTeam(client: PostgresRouter): Promise<HogFunctionCache> {
-    const items = (
-        await client.query<HogFunctionType>(
-            PostgresUse.COMMON_READ,
-            `
-            SELECT ${HOG_FUNCTION_FIELDS.join(', ')}
-            FROM posthog_hogfunction
-            WHERE deleted = FALSE AND enabled = TRUE
-        `,
-            [],
-            'fetchAllHogFunctions'
-        )
-    ).rows
+    public async enrichWithIntegrations(items: HogFunctionType[]): Promise<void> {
+        const integrationIds: number[] = []
+
+        items.forEach((item) => {
+            item.inputs_schema?.forEach((schema) => {
+                if (schema.type === 'integration') {
+                    const input = item.inputs?.[schema.key]
+                    if (input && typeof input.value === 'number') {
+                        integrationIds.push(input.value)
+                    }
+                }
+            })
+        })
 
-    const cache: HogFunctionCache = {}
-    for (const item of items) {
-        if (!cache[item.team_id]) {
-            cache[item.team_id] = {}
+        if (!items.length) {
+            return
         }
 
-        cache[item.team_id][item.id] = item
-    }
-
-    return cache
-}
-
-async function fetchEnabledHogFunctions(
-    client: PostgresRouter,
-    ids: HogFunctionType['id'][]
-): Promise<HogFunctionType[]> {
-    const items: HogFunctionType[] = (
-        await client.query(
-            PostgresUse.COMMON_READ,
-            `SELECT ${HOG_FUNCTION_FIELDS.join(', ')}
-                FROM posthog_hogfunction
-                WHERE id = ANY($1) AND deleted = FALSE AND enabled = TRUE`,
-            [ids],
-            'fetchEnabledHogFunctions'
+        const integrations: IntegrationType[] = (
+            await this.postgres.query(
+                PostgresUse.COMMON_READ,
+                `SELECT id, team_id, kind, config, sensitive_config
+                FROM posthog_integration
+                WHERE id = ANY($1)`,
```

Comment:
Do we want to add `and team_id = $2` here for safety when fetching by `id`?

Reply:
Nope. This can be called in such a way that lots of functions for different teams are being enriched (the next step takes care of ensuring only those for the function's team are added